src/posix/platform/radio.cpp

```diff
@@ -87,14 +87,28 @@ Radio::Radio(const char *aUrl)
 
 void Radio::Init(void)
 {
-    bool        resetRadio             = (mRadioUrl.GetValue("no-reset") == nullptr);
-    bool        restoreDataset         = (mRadioUrl.GetValue("ncp-dataset") != nullptr);
-    bool        skipCompatibilityCheck = (mRadioUrl.GetValue("skip-rcp-compatibility-check") != nullptr);
-    const char *parameterValue;
-    const char *region;
+    bool         resetRadio             = (mRadioUrl.GetValue("no-reset") == nullptr);
+    bool         restoreDataset         = (mRadioUrl.GetValue("ncp-dataset") != nullptr);
+    bool         skipCompatibilityCheck = (mRadioUrl.GetValue("skip-rcp-compatibility-check") != nullptr);
+    spinel_iid_t iidList[Spinel::kSpinelHeaderMaxNumIid];
+    const char  *parameterValue;
+    const char  *region;
 #if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
     const char *maxPowerTable;
 #endif
+    struct ot::Spinel::RadioSpinelCallbacks callbacks;
+
+    memset(&callbacks, 0, sizeof(callbacks));
+#if OPENTHREAD_CONFIG_DIAG_ENABLE
+    callbacks.mDiagReceiveDone  = otPlatDiagRadioReceiveDone;
+    callbacks.mDiagTransmitDone = otPlatDiagRadioTransmitDone;
+#endif // OPENTHREAD_CONFIG_DIAG_ENABLE
+    callbacks.mEnergyScanDone = otPlatRadioEnergyScanDone;
+    callbacks.mReceiveDone    = otPlatRadioReceiveDone;
+    callbacks.mTransmitDone   = otPlatRadioTxDone;
+    callbacks.mTxStarted      = otPlatRadioTxStarted;
```

Comment:
Curious, why did we change to set these callbacks from here? Is it needed for multipan change?

Reply:
If I understand the question correctly - this is similar to the previous question about the structure. This is needed to allow use of multiple interfaces in a single process, other interface may want to use own implementation of the handlers.